artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/LinkedListImpl.java

```diff
@@ -203,20 +223,31 @@ public void addSorted(E e) {
          // This would be an optimization for our usage.
          // avoiding scanning the entire List just to add at the end, so we compare the end first.
          if (comparator.compare(tail.val(), e) >= 0) {
+            logger.trace("addTail as compare is after tail {}", e);
             addTail(e);
             return;
          }
 
-         Node<E> fetching = head.next;
-         while (fetching.next != null) {
-            int compareNext = comparator.compare(fetching.next.val(), e);
-            if (compareNext <= 0) {
-               addAfter(fetching, e);
-               return;
+         if (lastAdd != null) { // as an optimization we check against the last add rather than always scan.
+            if (lastAdd.prev != null && lastAdd.prev.val() != null) {
+               if (comparator.compare(lastAdd.prev.val(), e) > 0 && comparator.compare(lastAdd.val(), e) < 0) {
+                  logger.trace("addABefore last element {}", e);
+                  addAfter(lastAdd.prev, e);
```

Comment:
addABefore has a typo, but more importantly the code then executes addAfter which makes it look wrong since the other log messages in this area refer to literal method names, though it isnt wrong since it passes the previous node. Also "last element" sounds a bit like it will be the tail, though clearly isnt here.

Something like "adding before most recently added element" might be clearer?